jax/_src/lax/parallel.py

```diff
@@ -586,10 +599,16 @@ def _all_to_all_translation_rule(c, x, *, split_axis, concat_axis, axis_name,
       x = xla.lower_fun(partial(lax.squeeze, dimensions=(split_axis,)), multiple_results=False)(c, x)
       return x
 
-def _all_to_all_transpose_rule(cts, axis_name, split_axis, concat_axis):
-  return (all_to_all(cts, axis_name=axis_name, split_axis=concat_axis, concat_axis=split_axis),)
+def _all_to_all_transpose_rule(cts, axis_name, split_axis, concat_axis, axis_index_groups):
+  return all_to_all(
+      cts,
+      axis_name=axis_name,
+      split_axis=concat_axis,
+      concat_axis=split_axis,
+      axis_index_groups=axis_index_groups)
```

Comment:
Note that this used to return a tuple, but doesn't do that anymore. That might explain why you're seeing surprising failures in the grad tests.

Reply:
I'm embarrassed by just a dumb mistake. I spent hours looking at my code without noticing it... (╯°□°）╯︵ ┻━┻

Good news is that everything works as expected now.

Reply:
Well I'm glad it works now 😄